### PR TITLE
Try enabling macOS ARM64 wheels

### LINKED ISSUE #832: Support for Apple's arm64 architecture (osx-arm64)
Hello, I get the following error when trying to install tables on a MacBook Pro with the M1 processor.  Part of the error shows PyTables supports ARM, but it failed to detect the CPU, so I'm guessing it's related to `osx-arm64`

`cpuinfo failed, assuming no CPU features: py-cpuinfo currently only works on X86 and some PPC and ARM CPUs.`

Here's the full error:

```
Collecting tables
  Using cached tables-3.6.1.tar.gz (4.6 MB)
    ERROR: Command errored out with exit status 1:
     command: /Users/ryan/miniforge3/envs/ft/bin/python3.8 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/6g/73g1_dms191f5pj1mc5n8byh0000gn/T/pip-install-6weqp8s9/tables_6efa87f1030c473590e79b8388430fc1/setup.py'"'"'; __file__='"'"'/private/var/folders/6g/73g1_dms191f5pj1mc5n8byh0000gn/T/pip-install-6weqp8s9/tables_6efa87f1030c473590e79b8388430fc1/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/6g/73g1_dms191f5pj1mc5n8byh0000gn/T/pip-pip-egg-info-zbtiyiw2
         cwd: /private/var/folders/6g/73g1_dms191f5pj1mc5n8byh0000gn/T/pip-install-6weqp8s9/tables_6efa87f1030c473590e79b8388430fc1/
    Complete output (12 lines):
    /var/folders/6g/73g1_dms191f5pj1mc5n8byh0000gn/T/H5closeb4hby9yj.c:2:5: error: implicit declaration of function 'H5close' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        H5close();
        ^
    1 error generated.
    cpuinfo failed, assuming no CPU features: py-cpuinfo currently only works on X86 and some PPC and ARM CPUs.
    * Using Python 3.8.6 | packaged by conda-forge | (default, Nov 27 2020, 18:57:00)
    * USE_PKGCONFIG: False
    * Found conda env: ``/Users/ryan/miniforge3/envs/ft``
    .. ERROR:: Could not find a local HDF5 installation.
       You may need to explicitly state where your local HDF5 headers and
       library can be found by setting the ``HDF5_DIR`` environment
       variable or by using the ``--hdf5`` command-line option.
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

### PR DESCRIPTION
**Pull request by cbrnr**
Supersedes #1049 (correct branch name). Fixes #832.

### CODE REVIEW

**Comment by cbrnr** [comment]
@avalentino hopefully this works!

**Comment by avalentino** [comment]
It seems that the run failed https://github.com/cbrnr/PyTables/actions/runs/6093682535
There is a problem with patching that could not be related to osx-arm.

**Comment by cbrnr** [comment]
Maybe those patches are for a previous version of hdf5? What do they do? Are they still necessary? Given that hdf5 supports macOS ARM64, it might be the case that everything works without the patches. I'll give it a try.

**Comment by cbrnr** [comment]
@avalentino unfortunately, it looks like you have to approve the workflows every time I push something new...

**Comment by avalentino** [comment]
> @avalentino unfortunately, it looks like you have to approve the workflows every time I push something new...

Just done, but please note that the job building wheels is only visible in the GHA of your fork

**Comment by cbrnr** [comment]
Thanks. It's not working though. Do we really need to build HDF5 from source when there is an [official binary for macOS ARM64](https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.14/hdf5-1.14.1/bin/unix/hdf5-1.14.1-2-Std-macos11m1_64-clang.tar.gz)?

**Comment by avalentino** [comment]
Not sure.
We need to check if the fact of building from sources is used to enforce specific build flags (e.g. the threading ones)
Probably for linux it is more relevant to ensure compatibility with the c runtime of the target linux wheel, but for mac I guess we can use a pre-build version of the HDF5 lib

**Comment by cbrnr** [comment]
Yes, since there is only one Mac ARM64 platform, I think this should be reasonable. It will certainly make the process a lot easier. Let me know when I can help (e.g. with testing).

**Comment by cbrnr** [comment]
FWIW, I just tried to build HDF5 on my Mac, and it worked out of the box. I tried the latest version 1.14.2, downloaded and unpacked the source, created a `build` directory and ran `cmake .. && make`. So at least compilation doesn't seem to be an issue (not sure how I can run tests).

**Comment by cbrnr** [comment]
Re the two sources, I can also build the other (CMake-hdf5-1.14.2) just fine. I think the difference is just the folder structure. Using `make -j8`, I can build the whole thing in about 10s.

**Comment by avalentino** [comment]
OK, then the patch could be a leftover from a previous attempt.
Please feel free to proceed as you feel it is more conveninet.

**Comment by cbrnr** [comment]
This process is a bit cumbersome, I'm basically trying to replicate what I'm doing locally, but it's taking ages on CI. Is there some way to work interactively? Or is there someone with a bit more experience than me? All that needs to be done is move the extracted files to the right location, and we should be good.

**Comment by avalentino** [comment]
I can only suggest to switch off temporarily all the CI jobs that are not relevant for your patch (including the wheels generation on non max platforms).

As a side note, if you add `pull_request:` to the triggers we should be able to see the tst results directly in the PR:

```
on:
  pull_request:
  pull:
    ...
```

**Comment by cbrnr** [comment]
Seems like it is working now, wheels for macOS ARM64 are available. Linux AARCH64 is still running, so I can't download the wheel (artifact) in order to test it locally. I assume that everything is fine if all tests pass?

Regarding the installation instructions, they still mention `brew install hdf5`. But with the wheel, I don't think this is necessary anymore, because all dependencies should be included (for all platforms that have binary wheels, which is basically *all* platforms). Therefore, I'd modify the installation instruction to just `pip install tables`.

**Comment by avalentino** [comment]
> Seems like [it is working now](https://github.com/cbrnr/PyTables/actions/runs/6107751002/job/16575396039), wheels for macOS ARM64 are available.

wonderful! thank you

> Linux AARCH64 is still running (this takes ages!), so I can't download the wheel (artifact) in order to test it locally (https://github.com/cbrnr/PyTables/actions/runs/6107751002).

yes, I know
Maybe at some point I shall move that task to a separate pipeline

> I assume that everything is fine if all tests pass?

yes, correct

> Regarding the installation instructions, they still mention brew install hdf5. But with the wheel, I don't think this is necessary anymore, because all dependencies should be included (for all platforms that have binary wheels, which is basically all platforms). Therefore, I'd modify the installation instruction to just pip install tables.

Thanks, very appreciated

**Comment by cbrnr** [comment]
> Maybe at some point I shall move that task to a separate pipeline

Yes, this would probably be a good idea, since the whole process takes more than 5 hours according to previous logs.

Once this has finished, I'll download the package and run the tests. Then I'll update the installation instructions, and hopefully this will be it. I'll report back!

**Comment by cbrnr** [comment]
Doesn't look too promising:

```
❯ python -m tables.tests.test_all 
Traceback (most recent call last):
  File "<frozen runpy>", line 189, in _run_module_as_main
  File "<frozen runpy>", line 112, in _get_module_details
  File "/Users/clemens/Downloads/artifact/.direnv/python-3.11/lib/python3.11/site-packages/tables/__init__.py", line 41, in <module>
    f"I looked for \"{', '.join(blosc2_search_paths)}\"")
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: sequence item 2: expected str instance, NoneType found
```

**Comment by cbrnr** [comment]
But the library is inside the package folder (in `site-packages/tables`):

```
❯ ls
__pycache__                        exceptions.py                           lrucacheextension.cpython-311-darwin.so
misc                               expression.py                           lrucacheextension.pxd
nodes                              file.py                                 lrucacheextension.pyx
scripts                            filters.py                              node.py
tests                              flavor.py                               parameters.py
__init__.py                        group.py                                path.py
_comp_bzip2.cpython-311-darwin.so  hdf5extension.cpython-311-darwin.so     registry.py
_comp_bzip2.pyx                    hdf5extension.pxd                       req_versions.py
_comp_lzo.cpython-311-darwin.so    hdf5extension.pyx                       table.py
_comp_lzo.pyx                      idxutils.py                             tableextension.cpython-311-darwin.so
_version.py                        index.py                                tableextension.pyx
array.py                           indexes.py                              undoredo.py
atom.py                            indexesextension.cpython-311-darwin.so  unimplemented.py
attributeset.py                    indexesextension.pyx                    utils.py
carray.py                          leaf.py                                 utilsextension.cpython-311-darwin.so
conditions.py                      libblosc2.dylib                         utilsextension.pxd
definitions.pxd                    link.py                                 utilsextension.pyx
description.py                     linkextension.cpython-311-darwin.so     vlarray.py
earray.py                          linkextension.pyx
```

**Comment by avalentino** [comment]
The problem is in the following test for early detection of missing libraries.
... and there is also a `None` in `blosc2_search_paths` that would need to be handled.

```python
# Load the blosc2 library, and if not found in standard locations,
# try this directory (it should be automatically copied in setup.py).
current_dir = os.path.dirname(__file__)
platform_system = platform.system()
blosc2_lib_hardcoded = "libblosc2"
if platform_system == "Linux":
    blosc2_lib_hardcoded += ".so"
elif platform_system == "Darwin":
    blosc2_lib_hardcoded += ".dylib"
else:
    blosc2_lib_hardcoded += ".dll"
blosc2_found = False
blosc2_search_paths = [blosc2_lib_hardcoded,
                       os.path.join(current_dir, blosc2_lib_hardcoded),
                       find_library("blosc2")]
for blosc2_lib in blosc2_search_paths:
    if blosc2_lib:
        try:
            cdll.LoadLibrary(blosc2_lib)
        except OSError:
            pass
        else:
            blosc2_found = True
            break
if not blosc2_found:
    raise RuntimeError("Blosc2 library not found. "
                       f"I looked for \"{', '.join(blosc2_search_paths)}\"")
```

**Comment by cbrnr** [comment]
OK, the second entry in `blosc2_search_paths` contains the correct lib, but it wasn't built for `arm64` unfortunately:

```
> cdll.LoadLibrary(blosc2_search_paths[1])
*** OSError: dlopen(/Users/clemens/Projects/PyTables/.direnv/python-3.11/lib/python3.11/site-packages/tables/libblosc2.dylib, 0x0006): tried: '/Users/clemens/Projects/PyTables/.direnv/python-3.11/lib/python3.11/site-packages/tables/libblosc2.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/Users/clemens/Projects/PyTables/.direnv/python-3.11/lib/python3.11/site-packages/tables/libblosc2.dylib' (no such file), '/Users/clemens/Projects/PyTables/.direnv/python-3.11/lib/python3.11/site-packages/tables/libblosc2.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64'))
```

Do you know what I need to change in order to build all dependencies for the correct arch?

**Comment by avalentino** [comment]
Dear @cbrnr I'm traveling this week so I cannot help too much.
I'm pretty sure that blosc2 provides wheels for OSX-ARM so probably something went wrong during the installation of dependencies.
For sure @FrancescAlted can give a better advice about how to install blosc2 properly on OSX-ARM.

**Comment by cbrnr** [comment]
Apparently, the package tries to use the included library, which was built for the wrong arch. The problem is that the CI job installs it with `pip install blosc2`, but this fetches the wrong arch (see e.g. https://github.com/cbrnr/PyTables/actions/runs/6113073091/job/16591825966#step:8:187). It's probably just some env variable missing...

**Comment by FrancescAlted** [comment]
Hi @cbrnr .  Thanks for tackling this work.  Probably setting this [CMAKE_OSX_ARCHITECTURES](https://cmake.org/cmake/help/latest/variable/CMAKE_OSX_ARCHITECTURES.html) envar to `arm64`  in the Mac/arm64 section would help.

**Comment by FrancescAlted** [comment]
[CIBW_ARCHS_MACOS: "arm64"](https://github.com/Blosc/blosc2_btune/blob/main/.github/workflows/cibuildwheels.yml#L98) may also be helpful.

**Comment by cbrnr** [comment]
Thanks, I'll try both!

**Comment by cbrnr** [comment]
It's not working: https://github.com/cbrnr/PyTables/actions/runs/6157538170/job/16708471840#step:8:189

Somehow, all packages installed by `before_build` are `x86_64`...

**Comment by FrancescAlted** [comment]
Interesting. I am getting out of ideas, but can you try with something similar to:

```
         CIBW_BEFORE_BUILD: >
            python -m pip install -U setuptools pip wheel &&
            python -m pip install -r requirements.txt &&
            CIBW_ARCHS_MACOS=arm64 python -m pip install blosc2
```

**Comment by cbrnr** [comment]
OK, I tried with:

```
          CIBW_BEFORE_BUILD: >
            export CIBW_ARCHS_MACOS=arm64 &&
            python -m pip install -U setuptools pip wheel &&
            python -m pip install -r requirements.txt &&
            python -m pip install blosc2
```

I'll let you know how that goes.

**Comment by cbrnr** [comment]
It's not working: https://github.com/cbrnr/PyTables/actions/runs/6157830380/job/16709389903#step:8:147

I'm also running out of ideas (and time).

**Comment by FrancescAlted** [comment]
FYI, we are making rapid progress towards a 3.9.0 release in branch https://github.com/PyTables/PyTables/tree/releases/v3.9.0.  We are planning to do a release early next week.  In case anyone want to try again with this, that would be a good timeframe for including arm64 support.

**Comment by cbrnr** [comment]
I'd love to help, but I'm out of ideas. Just in case we're not using cibuildwheel correctly, @henryiii do you know why our macOS ARM64 job is building/installing dependencies as x86?

**Comment by cbrnr** [comment]
FYI, the changes in README.rst are not really related, so they could be moved/removed if you want.

**Comment by cbrnr** [comment]
@avalentino @FrancescAlted I'm trying to solve this issue, here is a summary of what I think is the problem:

- The build process *seems* to be working on macOS ARM64, because it generates corresponding wheels.
- However, `import tables` fails in 
`tables/__init__.py`, because it does not find `blosc2`.
- The `tables` wheel does contain a dylib though (replace `$SITEPACKAGES` with the actual site-packages directory):
  ```
  ❯ ls $SITEPACKAGES/tables/lib*
  libblosc2.dylib
  ```
- However, this library is `x86_64` instead of `arm64`, that's probably why it cannot be found:
  ```
  ❯ file $SITEPACKAGES/tables/libblosc2.dylib            
  libblosc2.dylib: Mach-O 64-bit dynamically linked shared library x86_64
  ```
- Interestingly, `pip install blosc2` does have a macOS ARM64 wheel, but it does not contain a file named `libblosc2.dylib`. Instead, it is called `blosc2_ext.cpython-311-darwin.so`:
  ```
  ❯ ls     
  __pycache__  __init__.py  blosc2_ext.cpython-311-darwin.so  core.py  info.py  ndarray.py  schunk.py  version.py
  ```
- This library is the correct architecture:
  ```
  ❯ file blosc2_ext.cpython-311-darwin.so
  blosc2_ext.cpython-311-darwin.so: Mach-O 64-bit bundle arm64
  ```

So maybe PyTables is just not detecting the blosc2 library because of the name? WDYT? Any recommendations on how to proceed?

**Comment by avalentino** [comment]
Dear @cbrnr thanks a lot for working on this.
I think that `libblosc2.dylib` and `blosc2_ext.cpython-311-darwin.so` are two different things:
* one is the blosc2 C library,
* the other one should be the module containing the Python API (bindings)

I think that the assumption (in the PyTables `setup.py`) is that the blosc2 wheel includes both and the Python binding directly use the C library.
If it is not the case, IMHO this is the root of the issue.

**Comment by cbrnr** [comment]
Thanks for clarifying. PyTables is explicitly searching for the C lib, and the PyTables macOS ARM64 wheel does contain `libblosc2.dylib`, but like I said, this is the wrong architecture.

The Python `blosc2` package should also contain the C sources and library, so I it can be used as a drop-in replacement for the C lib. Indeed, the Python `lib` directory contains the correct library:

```
❯ ls
cmake  python3.11  libblosc2.2.10.5.dylib  libblosc2.2.dylib  libblosc2.a  libblosc2.dylib

❯ file libblosc2.dylib                 
libblosc2.dylib: Mach-O 64-bit dynamically linked shared library arm64
```

So maybe PyTables is just looking in the wrong place?

**Comment by cbrnr** [comment]
Here's the content of `blosc2-2.2.9-cp311-cp311-macosx_11_0_arm64.whl` (the latest version available on PyPI):

```
❯ unzip -l blosc2-2.2.9-cp311-cp311-macosx_11_0_arm64.whl                                              
Archive:  blosc2-2.2.9-cp311-cp311-macosx_11_0_arm64.whl
  Length      Date    Time    Name
---------  ---------- -----   ----
     3677  10-05-2023 16:57   blosc2-2.2.9.dist-info/RECORD
      104  10-05-2023 16:56   blosc2-2.2.9.dist-info/WHEEL
        7  10-05-2023 16:56   blosc2-2.2.9.dist-info/top_level.txt
     1628  10-05-2023 16:56   blosc2-2.2.9.dist-info/LICENSE.txt
     9113  10-05-2023 16:56   blosc2-2.2.9.dist-info/METADATA
    19739  10-05-2023 16:37   blosc2-2.2.9.data/data/include/b2nd.h
    94187  10-05-2023 16:37   blosc2-2.2.9.data/data/include/blosc2.h
     1041  10-05-2023 16:37   blosc2-2.2.9.data/data/include/blosc2/codecs-registry.h
     1328  10-05-2023 16:37   blosc2-2.2.9.data/data/include/blosc2/blosc2-stdio.h
     2583  10-05-2023 16:37   blosc2-2.2.9.data/data/include/blosc2/blosc2-common.h
      796  10-05-2023 16:37   blosc2-2.2.9.data/data/include/blosc2/tuners-registry.h
     1811  10-05-2023 16:37   blosc2-2.2.9.data/data/include/blosc2/blosc2-export.h
      911  10-05-2023 16:37   blosc2-2.2.9.data/data/include/blosc2/filters-registry.h
  1526430  10-05-2023 16:56   blosc2-2.2.9.data/data/lib/libblosc2.2.dylib
  1912712  10-05-2023 16:56   blosc2-2.2.9.data/data/lib/libblosc2.a
  1526430  10-05-2023 16:56   blosc2-2.2.9.data/data/lib/libblosc2.dylib
  1526430  10-05-2023 16:56   blosc2-2.2.9.data/data/lib/libblosc2.2.10.5.dylib
     1405  10-05-2023 16:55   blosc2-2.2.9.data/data/lib/cmake/Blosc2/Blosc2Targets-release.cmake
     2765  10-05-2023 16:55   blosc2-2.2.9.data/data/lib/cmake/Blosc2/Blosc2ConfigVersion.cmake
     4497  10-05-2023 16:55   blosc2-2.2.9.data/data/lib/cmake/Blosc2/Blosc2Targets.cmake
     2416  10-05-2023 16:55   blosc2-2.2.9.data/data/lib/cmake/Blosc2/Blosc2Config.cmake
      278  10-05-2023 16:37   blosc2-2.2.9.data/data/lib/cmake/Blosc2/Modules/FindLZ4.cmake
     1097  10-05-2023 16:37   blosc2-2.2.9.data/data/lib/cmake/Blosc2/Modules/toolchain-armsf.cmake
      204  10-05-2023 16:37   blosc2-2.2.9.data/data/lib/cmake/Blosc2/Modules/FindZSTD.cmake
     2446  10-05-2023 16:37   blosc2-2.2.9.data/data/lib/cmake/Blosc2/Modules/FindZLIB_NG.cmake
     2091  10-05-2023 16:37   blosc2-2.2.9.data/data/lib/cmake/Blosc2/Modules/FindIPP.cmake
      995  10-05-2023 16:37   blosc2-2.2.9.data/data/lib/cmake/Blosc2/Modules/toolchain-armhf.cmake
      959  10-05-2023 16:37   blosc2-2.2.9.data/data/lib/cmake/Blosc2/Modules/toolchain-aarch64.cmake
     2210  10-05-2023 16:37   blosc2-2.2.9.data/data/lib/cmake/Blosc2/Modules/FindSIMD.cmake
    32195  10-05-2023 16:56   blosc2/schunk.py
       22  10-05-2023 16:56   blosc2/version.py
     4768  10-05-2023 16:56   blosc2/__init__.py
    47099  10-05-2023 16:56   blosc2/core.py
    21432  10-05-2023 16:56   blosc2/ndarray.py
  2000461  10-05-2023 16:56   blosc2/blosc2_ext.cpython-311-darwin.so
     2178  10-05-2023 16:56   blosc2/info.py
---------                     -------
  8758445                     36 files
```

**Comment by cbrnr** [comment]
If I copy the lib from the `blosc2` package into `site-packages/tables` (thereby overwriting the existing x86_64 dylib), I get a different error:

```
❯ python -c "import tables"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/clemens/Downloads/artifact/.direnv/python-3.11/lib/python3.11/site-packages/tables/__init__.py", line 44, in <module>
    from .utilsextension import get_hdf5_version as _get_hdf5_version
ImportError: dlopen(/Users/clemens/Downloads/artifact/.direnv/python-3.11/lib/python3.11/site-packages/tables/utilsextension.cpython-311-darwin.so, 0x0002): symbol not found in flat namespace '_b2nd_create_ctx'
```

**Comment by FrancescAlted** [comment]
Yes, it looks like the blosc2 shared library is copied from the wrong wheel (x86_64 instead of arm64).  That's strange.

**Comment by henryiii** [comment]
cibuildwheel 2.16.5 supports it (released yesterday).

**Comment by maxnoe** [comment]
I think you need to change the images (matrix.os) to `macos-13` for the x86 arch and `macos-14` for the arm arch.

See: https://cibuildwheel.pypa.io/en/stable/setup/#github-actions

and: https://github.com/actions/runner-images?tab=readme-ov-file#available-images

**Comment by avalentino** [comment]
@maxnoe, if @cbrnr agrees I think that you should be able to push changes directly on the PR branch.

**Comment by cbrnr** [comment]
Of course, please go ahead! Note that this PR should also be rebased if you have time.

**Comment by maxnoe** [comment]
That would require @cbrnr giving me write access to their fork

**Comment by cbrnr** [comment]
@maxnoe are you sure that this is not already correct? I was using `macos-latest`, which is the ARM64 version of macOS 14. So if anything, the Intel architecture might have been wrong, so I've changed that to `macos-latest-large` now, which should be macOS 14 on x86.

**Comment by cbrnr** [comment]
Actions are available in my fork: https://github.com/cbrnr/PyTables/actions

**Comment by cbrnr** [comment]
I'm not sure if this is a joke, but I cannot run the jobs in my fork ("The job was not started because recent account payments have failed or your spending limit needs to be increased. Please check the 'Billing & plans' section in your settings.")...

**Comment by maxnoe** [comment]
This rollout happens in stages, see the comment below the list, so you cannot really be sure what macos-latest is right now. For our CI, it was still MacOS 12 using x86,
you can check at the top of the log

**Comment by avalentino** [comment]
Apparently the free plan has a limit of 2000minutes per month.
Honestly I didn't know
By the way, if you rebase on the current master or cherry-pich 62b0005 you should be able to run GHA on the PR

**Comment by maxnoe** [comment]
The -large images are not free

**Comment by maxnoe** [comment]
You can see in your logs that also for you, macos-latest is still macos-12:

> Operating System
  macOS
  12.7.4
  21H1123
Runner Image
  Image: macos-12
  Version: 20240406.2

**Comment by cbrnr** [comment]
Are you looking at the right job? They are actually using the correct 13 and 14 versions: https://github.com/cbrnr/PyTables/actions/runs/8717689971/job/23913395483

**Comment by maxnoe** [comment]
@cbrnr Yes, it's working now (with macos-13 and macos-14), my comment was to explain why it didn't before (with macos-latest).

**Comment by padraic-padraic** [comment]
Downloaded and installed a wheel from the artifacts generated on @cbrnr 's fork. it installed cleanly and here's the result of `python3 -m tables.tests.test_all`:
```
(artifact) padraic@Padraics-MacBook-Air ~/D/artifact> python3 -m tables.tests.test_all
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
PyTables version:    3.9.3.dev0
HDF5 version:        1.14.3
NumPy version:       1.26.4
Numexpr version:     2.10.0 (not using Intel's VML/MKL)
Zlib version:        1.2.12 (in Python interpreter)
LZO version:         2.10 (Mar 01 2017)
BZIP2 version:       1.0.8 (13-Jul-2019)
Blosc version:       1.21.5 (2023-05-16)
Blosc compressors:   blosclz (2.5.1), lz4 (1.9.4), lz4hc (1.9.4), zlib (1.2.13), zstd (1.5.2)
Blosc filters:       shuffle, bitshuffle
Blosc2 version:      2.14.3 (2023-04-06)
Blosc2 compressors:  blosclz (2.5.3), lz4 (1.9.4), lz4hc (1.9.4), zlib (1.2.11.zlib-ng), zstd (1.5.5)
Blosc2 filters:      shuffle, bitshuffle
Python version:      3.11.9 (main, Apr 16 2024, 10:13:46) [Clang 15.0.0 (clang-1500.3.9.4)]
Platform:            macOS-14.4.1-arm64-arm-64bit
Byte-ordering:       little
Detected cores:      8
Default encoding:    utf-8
Default FS encoding: utf-8
Default locale:      (None, UTF-8)
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
Performing only a light (yet comprehensive) subset of the test suite.
If you want a more complete test, try passing the --heavy flag to this script
(or set the 'heavy' parameter in case you are using tables.test() call).
The whole suite will take more than 4 hours to complete on a relatively
modern CPU and around 512 MB of main memory.
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
.........................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................ss....................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................ssssssssssssssssssssssssssssssss........................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................sssss...............................................................................................................................ssssssssss..........ssssssssss...............................................................................................................................................................ssssss......................................................................................................................................................................................................................s................................................................................................................................................................................................................................................................................................s...s.....................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
----------------------------------------------------------------------
Ran 6799 tests in 115.992s

OK (skipped=68)
```

**Comment by cbrnr** [comment]
@padraic-padraic can you confirm the architecture is indeed `arm64` using the steps outlined in my [previous comment](https://github.com/PyTables/PyTables/pull/1050#issuecomment-1801192852)?

**Comment by padraic-padraic** [comment]
> @padraic-padraic can you confirm the architecture is indeed `arm64` using the steps outlined in my [previous comment](https://github.com/PyTables/PyTables/pull/1050#issuecomment-1801192852)? I'm asking because previously the package also built and I could install it, but it was `x86_64` (and therefore running in Rosetta2).

```(artifact) padraic@Padraics-MacBook-Air ~/D/a/.v/l/p/s/tables> file *.dylib
libblosc2.dylib: Mach-O 64-bit dynamically linked shared library arm64```

Looks good!

**Comment by maxnoe** [comment]
The log clearly shows arm64:

 Building wheel for tables (pyproject.toml): finished with status 'done'
    Created wheel for tables: filename=tables-3.9.3.dev0-cp310-cp310-macosx_11_0_arm64.whl size=3588532 sha256=dfae86d86c6c45465e809cc89360692badbe13c72d64df6b67243ae6fc7e8859

**Comment by cbrnr** [comment]
Great! So I guess this is finally almost ready. I'll integrate the missing changes from `master` and then this should be good to go.

Do I need to add a changelog entry or is this done automatically?

**Comment by cbrnr** [comment]
What about the failing NumPy>=2 job though?

File: README.rst [hunk 1]
```diff
@@ -55,94 +55,78 @@ but rather as a teammate. If you want to work with large datasets of
...
 optional.
 
-Installation
-------------
-
-1. Make sure you have HDF5 version 1.10.5 or above.
-
-   On OSX you can install HDF5 using `Homebrew <http://brew.sh>`_::
+Make sure you have HDF5 version 1.10.5 or above. On macOS, you can
+install HDF5 with `Homebrew <http://brew.sh>`_::
```
[the remark this answers is not in the document]
Comment (cbrnr) [reply README.rst:109]: OK!

**Review by avalentino — APPROVED** [review]

**Comment by avalentino** [comment]
Thanks a lot to everybody for the precious help :)

**Comment by cbrnr** [comment]
May I ask when the next release is planned?

**Comment by avalentino** [comment]
I still do not have a plan at the moment.
I would like to have numpy2 compatibility in the next release, but I still need some time to implement it.

**Pull request merged**